modules/dr-recover-lost-control-plane-hosts.adoc

```diff
@@ -0,0 +1,301 @@
+// Module included in the following assemblies:
+//
+// * disaster_recovery/scenario-1-infra-recovery.adoc
+
+[id="dr-scenario-1-recover-master-hosts_{context}"]
+= Recovering from lost master hosts
+
+////
+TODO:
+ - Wording on a few steps.
+ - Add a final result
+////
+
+Follow these steps to recover from the loss of one or more master hosts.
+
+.Prerequisites
+
+* Access to the cluster as a user with the `cluster-admin` role.
+* SSH access to a remaining master host.
+
+.Procedure
+
+. Restore etcd quorum on the remaining master.
++
+[NOTE]
+====
+This step is only necessary if you have had a majority of your masters fail. You can skip this step if you have a majority of your masters still available.
+====
++
+.. Access the remaining master host.
+
+.. Set the `INITIAL_CLUSTER` variable to the list of members in the format of `<name>=<url>`. This variable will be passed to the restore script, and in this procedure, it is assumed that there is only a single member at this time.
++
+----
+[core@ip-10-0-143-125 ~]$ source /run/etcd/environment; export INITIAL_CLUSTER="etcd-member-$(hostname).$(hostname -d)=https://${ETCD_DNS_NAME}:2380"
+----
+
+.. Run the `etcd-snapshot-restore.sh` script.
++
+Pass in two parameters to the `etcd-snapshot-restore.sh` script: the path to a backed up etcd snapshot file and list of members, which is defined by the `INITIAL_CLUSTER` variable.
++
+// TODO: Is this snapshot file still optional? Wouldn't the parameters be out of order then if you didn't include it? Question out to Sam.
++
```

Comment:
Well in some cases the local db might be your only option. I agree that default should be an explicit snapshot path. If they want to use the local db they can handle that on their own.

Reply:
So to be clear the snapshot path should NOT be optional.

Reply:
So if the recommended way is to use a saved snapshot, should we go ahead and list that as a prereq (like how we do for scenario 2 here [1]).

[1] http://file.rdu.redhat.com/~ahoffer/2019/disaster-recovery/disaster_recovery/scenario-2-restoring-cluster-state.html

Reply:
yes